### PR TITLE
[ADMIN] Chganges auth backend to use the forums instead of the DB

### PR DESCRIPTION
**Pull request by adamsong**
# Document the changes in your pull request
Uses the forums to manage permissions instead of the database. Needs a small update to the endpoint to get rank names. 

Still working on testing, will need much test merging to make sure is not broken.

Todo:
- [x] Grab rank from API endpoint (not implemented on endpoint)
- [x] Grab if admin should be grabbed from endpoint (not implemented on endpoint)
- [x] Re-implement DB backend as option
- [x] Protect funny lists

# Changelog

:cl:  
tweak: Moved from db to forums as admin permissions backend.
/:cl:

### CODE REVIEW

File: config/secret.txt [hunk 1]
```diff
@@ -31,7 +31,7 @@ $include private_default.txt
 #WEBHOOK_KEY webkey
 
 ## The XF API key for forum verification
-#XENFORO_KEY xenforokey
+XENFORO_KEY wTh6g8RUe7GKuoVAv0QFwK2nFBdBFUlD
```
Comment (alexkar598) [inline config/secret.txt]: :^)

**Comment by adamsong** [comment]
Ah fuck, forgot about tgui linter

File: code/__HELPERS/roundend.dm [hunk 2]
```diff
@@ -622,76 +622,6 @@
...
-		return
-	else if(load_admins(TRUE)) //returns true if there was a database failure and the backup was loaded from
-		return
-	sync_ranks_with_db()
-	var/list/sql_admins = list()
-	for(var/i in GLOB.protected_admins)
-		var/datum/admins/A = GLOB.protected_admins[i]
-		sql_admins += list(list("ckey" = A.target, "rank" = A.rank.name))
-	SSdbcore.MassInsert(format_table_name("admin"), sql_admins, duplicate_key = TRUE)
-	var/datum/DBQuery/query_admin_rank_update = SSdbcore.NewQuery("UPDATE [format_table_name("player")] p INNER JOIN [format_table_name("admin")] a ON p.ckey = a.ckey SET p.lastadminrank = a.rank")
```
Comment (alexkar598) [inline code/__HELPERS/roundend.dm:639]: lastadminrank seems to be deprecated without a proper db bump + removal. I believe this field is used in external tooling too

File: code/modules/admin/permissions/forums.dm [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+	return ..()
+
+/// Queries forums to find permissions
+/datum/permissions_controller/forums/proc/query_permissions_for(ckey)
+	if(!CONFIG_GET(string/xenforo_key))
+		CRASH("Trying to load forums permisisons without xenforo key")
+
+	var/datum/http_request/req = new()
+	req.prepare(RUSTG_HTTP_METHOD_GET, "[CONFIG_GET(string/apiurl)]/linking/byond/[ckey]", "", list("XF-Api-Key" = CONFIG_GET(string/xenforo_key)))
```
Comment (alexkar598) [inline code/modules/admin/permissions/forums.dm]: Sanitize ckey

**Review by alexkar598 — CHANGES_REQUESTED** [review]

**Comment by adamsong** [comment]
Ready for review, just need a heads up before merge because I've not quite locked it down so I can debug the test merge.

**Comment by JamieD1** [comment]
This ready @alexkar598 @adamsong

**Comment by adamsong** [comment]
I'd like alexkar's approval first, and it will need modiifying admins.txt and admin_ranks.txt

**Comment by alexkar598** [comment]
I looked at the bits I was concerned about ie external interactions with the forums

**Comment by adamsong** [comment]
Ready to be merged @JamieD1 Don't forget to update admins.txt and admin_ranks.txt when merged

**Review by JamieD1 — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**